brian2/codegen/translation.py

```diff
@@ -211,6 +210,10 @@ def make_statements(code, variables, dtype, optimise=True, blockname=''):
     describing how the statement can be reformulated as a sequence of if/then
     statements. Calls `~brian2.codegen.optimisation.optimise_statements`.
     '''
+    cache_key = (code, _hashable(variables), dtype, optimise, blockname)
+    if cache_key in _make_statements_cache:
+        return _make_statements_cache[cache_key]
+
```

Comment:
Ok, I did this (I'll push the commit in a minute) -- we still have to be careful whenever we add arguments and think about how they relate to the caching, but I think the most common issue that we could have is that the cache no longer works.